File: Golang_src/easy/singleNumber_136.go
package easy

// Use bit manipulation:
// Time Complexity: O(n)
// Space Complexity:O(1)
// Runtime: 16 ms, faster than 56.29%
// Memory Usage: 6 MB, less than 73.88%
func singleNumber(nums []int) int {
	result := 0
	for _, v := range nums {
		result ^= v
	}

	return result
}

File: C_src/Easy/136. Single Number/SingleNumber_136.c
#include <stdio.h>
#include <stdlib.h>

// Use bit manipulation
// Time Complexity: O(n)
// Space Complexity:O(1)
// Runtime: 20 ms, faster than 19.21%
// Memory Usage: 8.1 MB, less than 63.64%
int singleNumber(int *nums, int numsSize)
{
    int i;
    int result = 0;
    for (i = 0; i < numsSize; i++)
    {
        result ^= nums[i];
    }
    return result;
}